src/hooks/usePermit/usePermit.ts

```diff
@@ -0,0 +1,66 @@
+import { useCallback } from 'react'
+import { DAI_TOKENS, MAX_UINT256, NONCES_FN, SUPPORTED_TOKENS } from './constants'
+import { call, signData } from './rpc'
+import type { TDaiPermitMessage, TERC2612PermitMessage, TRSVResponse, TUsePermitOptions } from './types'
+import { addZeros, createTypedDaiData, createTypedERC2612Data, getDomain, isTokenExists } from './utils'
+
+const usePermit = (options: TUsePermitOptions) => {
+  const { provider, token, spender, owner, chainId, deadline } = options
+
+  const signDaiPermit = useCallback(async () => {
+    const findToken = isTokenExists(DAI_TOKENS, { address: token, chainId })
+    const { noncesFn } = findToken || { }
+
+    const message: TDaiPermitMessage = {
+      holder: owner,
+      spender,
+      nonce: await call(provider, token, `${noncesFn || NONCES_FN}${addZeros(24)}${owner.slice(2)}`),
+      expiry: deadline || MAX_UINT256,
+      allowed: true
+    }
+
+    const domain = await getDomain(provider, token, chainId)
+    const typedData = createTypedDaiData(message, domain)
+    console.log({ typedData, message, domain })
+    const sig = await signData(provider, owner, typedData)
+
+    return { ...sig, ...message }
+  }, [provider, token, spender, owner, chainId, deadline])
+
+  const signERC2612Permit = useCallback(async (): Promise<TERC2612PermitMessage & TRSVResponse> => {
+    const message: TERC2612PermitMessage = {
+      owner,
+      spender,
+      value: MAX_UINT256,
```

Comment:
You forget to add `deadline` here

Reply:
But I think it's a part of the general problem: I would suggest to separate logic for dai and erc2612 permit to one function and use it twice

Reply:
In DAI we use `expiry`, I added it